hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmLifecycleConfiguration.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.om.helpers;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.ozone.OzoneConsts;
+import org.apache.hadoop.ozone.audit.Auditable;
+import org.apache.hadoop.ozone.om.exceptions.OMException;
+
+/**
+ * A class that encapsulates lifecycle configuration.
+ */
+public class OmLifecycleConfiguration extends WithObjectID
+    implements Auditable {
+
+  // Ref: https://docs.aws.amazon.com/AmazonS3/latest/userguide/intro-lifecycle-rules.html#intro-lifecycle-rule-id
+  public static final int LC_MAX_RULES = 1000;
+  private String volume;
+  private String bucket;
+  private String owner;
```

Comment:
Based on latest design document, owner field is removed.

Reply:
How to keep the owner? This can be used to confirm who created the Lifecycle, the audit log may be overwritten

Reply:
Owner will be fetched from the BucketInfo object.  Anyway we will update the modification time of Bucket, so we avoid saving duplicate owner info in Lifecycle configuration.

Reply:
However, this owner refers to the user who created the LifecycleConfiguration instead of the bucket owner. This user may be the owner of the bucket, or one of the administrators. If it is a Ranger, it may be a user with ALL permissions. Without this we may not know exactly who added this LifecycleConfiguration.

There are not many LifecycleConfigurations, so this field should not cause space problems

Reply:
I see. I know we have owner concept for an object, if we consider LifecycleConfiguration as part of Bucket information, we don't need to have a separate owner for the Configuration, right?

Reply:
OK, I removed the owner